File: backend/src/main/java/com/example/entities/FoodItemRequest.java
package com.example.entities;

import jakarta.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "food_item_request")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class FoodItemRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="request_id")
    private int requestId;

    @Column(name="item_name")
    private String itemName;

    @Column(name="item_description")
    private String description;

    private String status; // PENDING, APPROVED, REJECTED

    @ManyToOne
    @JoinColumn(name = "mess_id")
    private Mess messId;

    @ManyToOne
    @JoinColumn(name="sub_category_id")
    private SubCategory subCategory;
}

File: backend/Customer/Customer/Models/FoodItemRequest.cs
using System;
using System.Collections.Generic;

namespace Customer.Models;

public partial class FoodItemRequest
{
    public int RequestId { get; set; }

    public int MessId { get; set; }

    public string ItemName { get; set; } = null!;

    public string ItemDescription { get; set; } = null!;

    public string Status { get; set; } = null!;

    public int SubCategoryId { get; set; }

    public virtual Mess Mess { get; set; } = null!;

    public virtual SubCategory SubCategory { get; set; } = null!;
}
